File: BackEnd Api/model/marketmodel.js
const mongoose = require("mongoose");
const MarketModel = new mongoose.Schema(
  {
    userid:{
        type:String,
        required:true,
    },
    stockName: {
      type: String,
      required: true,
    },
    transactionType:{
        type: String,
        required: true,
      },
    quantity: {
      type: String,
      required: true,
    },
    amount: {
      type: String,
      required: true,
    },
  },
  {
    timestamps: true,
  }
);
const MarketRecords = mongoose.model("MarketRecords", MarketModel);
module.exports = MarketRecords;

File: stockapp/lib/models/market_model.dart
class MarketModel {
  String? sId;
  String? userid;
  String? stockName;
  String? transactionType;
  String? quantity;
  String? amount;
  String? createdAt;
  String? updatedAt;
  int? iV;

  MarketModel(
      {this.sId,
      this.userid,
      this.stockName,
      this.transactionType,
      this.quantity,
      this.amount,
      this.createdAt,
      this.updatedAt,
      this.iV});

  MarketModel.fromJson(Map<String, dynamic> json) {
    sId = json['_id'];
    userid = json['userid'];
    stockName = json['stockName'];
    transactionType = json['transactionType'];
    quantity = json['quantity'];
    amount = json['amount'];
    createdAt = json['createdAt'];
    updatedAt = json['updatedAt'];
    iV = json['__v'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['_id'] = this.sId;
    data['userid'] = this.userid;
    data['stockName'] = this.stockName;
    data['transactionType'] = this.transactionType;
    data['quantity'] = this.quantity;
    data['amount'] = this.amount;
    data['createdAt'] = this.createdAt;
    data['updatedAt'] = this.updatedAt;
    data['__v'] = this.iV;
    return data;
  }
}
